clients/src/main/java/org/apache/kafka/common/config/TopicConfig.java

```diff
@@ -67,13 +67,17 @@ public class TopicConfig {
         "(which consists of log segments) can grow to before we will discard old log segments to free up space if we " +
         "are using the \"delete\" retention policy. By default there is no size limit only a time limit. " +
         "Since this limit is enforced at the partition level, multiply it by the number of partitions to compute " +
-        "the topic retention in bytes.";
+        "the topic retention in bytes. Additionally, retention.bytes configuration " +
+        "operates independently of \"segment.ms\" and \"segment.byte\" configurations. " +
+        "Moreover, it triggers the expiration of active segment if retention.bytes is configured to zero.";
 
     public static final String RETENTION_MS_CONFIG = "retention.ms";
     public static final String RETENTION_MS_DOC = "This configuration controls the maximum time we will retain a " +
         "log before we will discard old log segments to free up space if we are using the " +
         "\"delete\" retention policy. This represents an SLA on how soon consumers must read " +
-        "their data. If set to -1, no time limit is applied.";
+        "their data. If set to -1, no time limit is applied. Additionally, retention.ms configuration " +
+        "operates independently of \"segment.ms\" and \"segment.byte\" configurations. " +
+        "Moreover, it triggers the expiration of active segment.";
```

Comment:
This sentence seems incomplete.

Reply:
Thanks for the feedback ! Do you think this is ok ?
```
Moreover, it triggers the expiration of active segment, 
segment expiration refers to the complete removal of segments from 
the partition once the retention.ms condition is satisfied.
```

Reply:
sounds good, @soarez , WDYT ?

Reply:
Sounds good to me too.